src/Native/Runtime/thread.cpp

```diff
@@ -979,6 +979,20 @@ EXTERN_C NOINLINE void FASTCALL RhpWaitForGC2(PInvokeTransitionFrame * pFrame)
     PalSetLastError(lastErrorOnEntry);
 }
 
+// Standard calling convention variant and actual implementation for RhpGcPoll
+EXTERN_C NOINLINE void FASTCALL RhpGcPoll2(PInvokeTransitionFrame* pFrame)
+{
+    Thread* pThread = ThreadStore::GetCurrentThread();
+    pFrame->m_pThread = pThread;
+
+    if (pThread->IsSuppressGcStressSet() || pThread->IsDoNotTriggerGcSet())
+    {
+        return;
+    }
+
```

Comment:
```suggestion
```
This should not be necessary.

Reply:
The `IsSuppressGcStressSet` part of the condition would actually cause GC stress to hang - once/if we decide to enable it.